csrc/class_datasource.cpp

```diff
@@ -32,7 +32,30 @@ JNIEXPORT jobject JNICALL Java_mapnik_Datasource_getParameters
 
 	for (mapnik::param_map::const_iterator iter=params.begin(); iter!=params.end(); iter++) {
 		jstring key=env->NewStringUTF(iter->first.c_str());
-		boost::apply_visitor(translate_parameter_visitor(env, paramobject, key), iter->second);
+		translate_parameter_visitor visitor(env, paramobject, key);
+		// TODO - The call to visit() does not compile on MSVC 2015 (error C2783).
```

Comment:
It doesn't compile on OS X, either. I'm not sure if I am even supposed to call `mapnik::value_type::visit` directly? It worked fine in another place.